api/events/schema.py

```diff
@@ -50,18 +44,31 @@ class Arguments:
     event = graphene.Field(Events)
 
     def mutate(self, info, **kwargs):
-        room_id = RoomSchedules().check_event_status(info, **kwargs)
-        cancelled_event = EventsModel(
-            event_id=kwargs['event_id'],
-            room_id=room_id,
-            event_title=kwargs['event_title'],
-            start_time=kwargs['start_time'],
-            end_time=kwargs['end_time'],
-            checked_in=False,
-            cancelled=True)
-        cancelled_event.save()
+        room_id, event = check_event_in_db(self, info, "cancelled", **kwargs)
+        if not event:
+            event = EventsModel(
+                event_id=kwargs['event_id'],
+                room_id=room_id,
+                event_title=kwargs['event_title'],
+                start_time=kwargs['start_time'],
+                end_time=kwargs['end_time'],
+                checked_in=False,
+                cancelled=True)
+            event.save()
+
+        return CancelEvent(event=event)
+
 
-        return CancelEvent(event=cancelled_event)
+def check_event_in_db(instance, info, event_check, **kwargs):
+    room_id = RoomSchedules().check_event_status(info, **kwargs)
+    event = EventsModel.query.filter_by(
+        start_time=kwargs['start_time'],
```

Comment:
@nakatuddesuzan why must we filter by `start_time`?

Reply:
Hello @joshuaocero , thanks for pointing this out. I don't think we need to filter by start_time. event_id alone can work since its unique.

Reply:
Hello @joshuaocero , on second thought, we need to query by start_time to cater for recurring events. If we don't include start_time we may end up canceling all events with the same `event_id`